junit5-support/src/main/kotlin/in/specmatic/test/SpecmaticJUnitSupport.kt

```diff
@@ -170,12 +46,30 @@ open class SpecmaticJUnitSupport {
 
         val testsNames = mutableListOf<String>()
         val partialSuccesses: MutableList<Result.Success> = mutableListOf()
-        val testReport: TestReport = TestReport()
+        private val reportConfiguration = getReportConfiguration()
+        private val openApiCoverageReportInput = OpenApiCoverageReportInput()
 
         @AfterAll
         @JvmStatic
         fun report() {
-            testReport.printReport()
+            val reportProcessors = listOf(OpenApiCoverageReportProcessor(openApiCoverageReportInput))
+            reportProcessors.forEach { it.process(reportConfiguration) }
+        }
+
+        private fun getReportConfiguration(): ReportConfiguration {
+            val reportConfiguration = reportConfigurationFromConfig()
+            val defaultFormatters = listOf(ReportFormatter(ReportFormatterType.TEXT, ReportFormatterLayout.TABLE))
+            val defaultReportTypes =
+                ReportTypes(apiCoverage = APICoverage(openAPI = APICoverageConfiguration(successCriteria = SuccessCriteria(0, 0, false))))
+            return when (reportConfiguration) {
+                null -> {
+                    logger.log("Report configuration found empty, hence configuring OpenAPI coverage report without any failure criteria ")
```

Comment:
Should we update this to "API coverage report configuration not found in specmatic.json, proceeding with API coverage report without success criteria"? This will be in line with the config field name "success criteria".

Reply:
Fixed.